### ISSUE TITLE
Wrong output when loading a finetuned mobileNetV3 model

### ISSUE THREAD

**Issue by hktime** [yalue/onnxruntime_go#33]
Thanks for your amazing work! 
I'm using a pretrained mobileNetV3 PyTorch model and finetuning it to work on a 2-category classification task. It had been exported as an onnx file and works well when using python onnxruntime library for inference.
But when it comes to golang inference, nothing works good, its prediction result is totally wrong. I'm wondering is there anything that I neglected.
onnxruntime Version: 1.16.1
Here is my code:
`

func predict(){

	imagePath := "test.jpg"
	f, err := os.Open(imagePath)
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	ort.SetSharedLibraryPath("./lib/libonnxruntime.dylib")
	err = ort.InitializeEnvironment()
	if err != nil {
		log.Fatal(err)
	}
	defer ort.DestroyEnvironment()

	inputData := prepareInput(f)
	inputShape := ort.NewShape(1, 3, 224, 224)
	inputTensor, err := ort.NewTensor(inputShape, inputData)
	if err != nil {
		log.Fatal(err)
	}
	defer inputTensor.Destroy()

	outputShape := ort.NewShape(1, 2)
	outputTensor, err := ort.NewEmptyTensor[float32](outputShape)
	if err != nil {
		log.Fatal(err)
	}
	defer outputTensor.Destroy()

	options, err := ort.NewSessionOptions()
	if err != nil {
		log.Fatal(err)
	}

	session, err := ort.NewAdvancedSession("./model/MobileNetV3_checkpoint.onnx",
		[]string{"input"}, []string{"output"},
		[]ort.ArbitraryTensor{inputTensor}, []ort.ArbitraryTensor{outputTensor}, options)
	defer session.Destroy()

	err = session.Run()

	outputData := outputTensor.GetData()
	}
`

**Comment by yalue** [comment]
That's odd.  I use python to validate the output of this library all the time; in fact, it's one of the main ways I test it.  Your code for setting up and running the network looks correct to me, and session.Run() should return an error if any of the shapes are wrong.  You do check that the `err` returned by `session.Run()` is nil, right?  It's possible that there is some error that you're missing when running the network that would be caught here.

Otherwise, my only guess is that somehow the image isn't being prepared correctly when you're calling `prepareInput` on your input image.  Have you checked that the array of floats returned by `prepareInput` is identical to the ones that are being passed to your network in python?

Sorry I can't be more help here!

**Comment by hktime** [comment]
Much appreciated for your kindly reply. I have followed your suggestion and double checked the error and `prepareInput` function. And it turns out to be the image pre-process procedure which should align with the procedures in python. I misused `imaging.Resize()` function, resulting in the array of floats that totally different with the tensor in python.

Your comment was truly enlightening, that really helps.
